src/Wrapper.js

```diff
@@ -311,6 +333,102 @@ class Wrapper extends React.PureComponent {
   }
 }
 
+const SwipeContainer = (() => {
+  let _progress = 0
+
+  const SwipeContainer = ({
+    children,
+    enabled,
+    menuPanelOpened,
+    onMenuPanelClose,
+    onMenuPanelOpen,
+    width,
+  }) => {
+    const oneThirdWindowWidth = width / 3
+
+    if (!enabled) {
+      return <Container>{children(Number(menuPanelOpened))}</Container>
+    }
+
+    return (
+      <Gesture passive={{ passive: true }}>
+        {({
+          delta: [xDelta, yDelta],
+          direction: [xDir, yDir],
+          down,
+          event,
+          initial: [xInitial],
+          xy: [x],
+        }) => {
+          if (!down && _progress > 0 && _progress < 1) {
+            _progress = !menuPanelOpened
+              ? _progress > THRESHOLD_PROGRESS
+                ? 1
+                : 0
+              : _progress < THRESHOLD_PROGRESS
+              ? 0
+              : 1
+            if (_progress === 1) {
+              setTimeout(() => {
+                _progress = 0
+                onMenuPanelOpen()
+              }, 1)
+            }
+            if (_progress === 0) {
+              setTimeout(() => {
+                onMenuPanelClose()
+              }, 1)
+            }
+            return <Container>{children(_progress)}</Container>
+          }
+
+          let progress = _progress || Number(menuPanelOpened)
+
+          if (
+            (progress > 0 && progress < 1) ||
+            (down &&
+              xDelta &&
+              yDelta > -THRESHOLD_DISTANCE &&
+              yDelta < THRESHOLD_DISTANCE &&
```

Comment:
I'm not sure how you accomplished such error, I've considered the behavior and checked against it, maybe that first click that seems of the edge spawned some inconsistencies:

![drag with vertical](https://user-images.githubusercontent.com/3072458/52854742-c3309900-311f-11e9-89bc-3a2ea525227c.gif)

Reply:
That’s really weird, in your capture `yDelta < THRESHOLD_DISTANCE` shouldn’t pass, since you are going below `10px`? But in any case, do we need this check at all?

Reply:
That's what the checks for `xDelta` and `xDir` are for, to capture "intent" if the user started scrolling horizontally and triggered the open/close, then it doesn't matter if they do vertical movement and the other way around, if they do vertical movement then the drag does not start.